woocommerce/woocommerce-gateway-stripe
# Fetch token pm_ id via get_token() rather than get_id()

## Changes proposed in this Pull Request:

While looking through some Stripe logs I stumbled on this error. 

<p align="center">
<img width="489" alt="Screenshot 2024-09-23 at 7 20 21 pm" src="https://github.com/user-attachments/assets/f3d780dd-8a44-4bd5-8949-d346284302f1">
</p> 

It turned out that this was being caused by a situation where a token's database ID was being passed to a stripe payment method request.

This PR fixes that. We didn't receive any reports of this as far as I can tell. It didn't cause any code issues (for us) because we pass the returned value to a `do_action()` we aren't using. 

## Testing instructions

1. Enable a redirect payment method that supports saved tokens like - iDeal, Cash App
1. Checkout `develop`
2. Go to **My Account > Payment Methods**
3. Attempt to add a new redirect payment method token.
4. once you return to the site, check your logs in the Stripe Dashboard
   - on `develop` you should see the error message above.
   - on this branch there should be no error.

---


**Post merge**
